### PR TITLE
[Feature] Support RetrieverRecall metric & Add ArcFace config

### PR DESCRIPTION
**Pull request by okotaku**
## Motivation

Related PR: https://github.com/open-mmlab/mmclassification/pull/1019

## Modification

Fix dataset.
Add arcface config.
Add RetrieverRecall metric.

## Checklist

**Before PR**:

- [x] Pre-commit or other linting tools are used to fix the potential lint issues.
- [x] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [x] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: mmcls/evaluation/metrics/retrieval.py [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+        >>> evaluator.process(data_samples)
+        >>> evaluator.evaluate(1000)
+        {'retrieval/Recall@1': 20.700000762939453,
+         'retrieval/Recall@5': 78.5999984741211}
+
+        Use in OpenMMLab configs:
+
+        .. code:: python
+
+            dict(type='RetrievalRecall', topk=(1, 5))
```
Comment (yingfhu) [inline mmcls/evaluation/metrics/retrieval.py:68]: `val/test_evaluator = dict(type='RetrievalRecall', topk=(1, 5))` should be better for understanding

File: mmcls/evaluation/metrics/retrieval.py [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+                    sample_target) in enumerate(zip(pred, target)):
+                sample_pred = np.array(to_tensor(sample_pred).cpu())
+                sample_target = np.array(to_tensor(sample_target).cpu())
+                recalls[i] = int(np.in1d(sample_pred[:k], sample_target).max())
+            results.append(recalls.mean() * 100)
+        return results
+
+
+def _format_pred(label, topk=None, is_indices=False):
+    """format various label to List[indeices]."""
```
Comment (yingfhu) [inline mmcls/evaluation/metrics/retrieval.py:189]: typo of indices

File: mmcls/evaluation/metrics/retrieval.py [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+    if is_indices:
+        assert isinstance(label, Sequence),  \
+                '`pred` must be Sequeue of indices when' \
```
Comment (yingfhu) [inline mmcls/evaluation/metrics/retrieval.py:192]: typo of sequence, and below as well

File: mmcls/evaluation/metrics/retrieval.py [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+        label = torch.from_numpy(label)
+    elif not isinstance(label, torch.Tensor):
+        raise TypeError(f'The pred must be type of torch.tensor, '
+                        f'np.ndarray or Sequeue but get {type(label)}.')
+    topk = topk if topk else label.size()[-1]
+    _, indices = label.topk(topk)
+    return indices
+
+
+def _format_target(label, is_indices=False):
```
Comment (yingfhu) [inline mmcls/evaluation/metrics/retrieval.py:212]: It seems unnecessary to have `_format` for both pred and target, could these be merged?

**Review by yingfhu — CHANGES_REQUESTED** [review]

File: mmcls/evaluation/metrics/retrieval.py (hunk 4 above)
Comment (okotaku) [reply mmcls/evaluation/metrics/retrieval.py:212]: @Ezra-Yu What do you think?

Comment (Ezra-Yu) [reply mmcls/evaluation/metrics/retrieval.py:212]: One of them has a topk argument, and they are logically different. I've tried this, but it's almost like merging the two functions together directly, without reducing the amount of code.

**Review by yingfhu — APPROVED** [review]
LGTM
